public/javascript/pump/socket.js

```diff
@@ -134,9 +149,34 @@
             }
         };
 
-        sock.onclose = function() {
-            // XXX: reconnect?
+        sock.onclose = function(e) {
             Pump.socket = null;
+            // Reconnect on broken connections, not for normal close
+            if (e.code !== 1000) {
+                sock.reconnect(e);
+            }
+        };
+
+        sock.reconnect = function(e) {
+            _socket.retryAttempts++;
+
+            if (_socket.retryTimer) {
+                clearTimeout(_socket.retryTimer);
+                _socket.retryTimer = null;
+            }
+
+            if (e.code === 2000 && _socket.retryAttempts > 10) {
+                // No more attempts if all transports failed
```

Comment:
Should probably show an error message to the user here

Reply:
Yes, should be show a message.

Reply:
Done 👍